source/_posts/2017-02-07-falsehoods-programmers-believe-about-language.markdown

```diff
@@ -0,0 +1,189 @@
+of pure happenstance and some languages (English is one) seem to have a
+preference for more shorter words while others prefer fewer longer words.
+
+
+## Translating a whole sentence of English is simple.
+
+A whole sentence as the smallest unit to be translated is better, but there is
+still plenty of room for ambiguity. Consider any sentence with "it" in it.
+In languages where every (or even some) noun is gendered, a translator needs to
+know what the antecedent is or at least its gender in order to correcly
+translate this word. If that antecedent is in a previous sentence, they'll have
+to guess and if they're wrong, the sentence will read as if it were written by a
+young child still learning the basics of grammar.
+
+
+## Templating sentences adds only a little complexity.
+
+It's common practice in the software industry to create templates out of
+sentences and substitute in short phrases or single words. You might see
+something like `"#{user.name} is in the #{user.location}."` where the user's
+name is, well, their name like "Ben Hamill" and the location might be one of
+"yard", "kitchen", "bathroom" or "bedroom". That seems reletively
+straightforward... in English.
+
+In languages with gendered nouns, those four location words might have different
+genders, meaning that the untemplated "the" would be different for different
+locations. If the translator lacks the context for what might go into that slot,
+they'll have no hope of being right all the time. For instance, in Spanish, "the
+kitchen" is "la cocina", but "the bathroom" is "el baño". If the translator has
+to pick either "el" _or_ "la", they're guaranteed to be wrong some of the time.
+
+
+## If I make sure to group things like articles in with nouns, that'll sort gendered languages.
+
+If you change the above example to `"#{user.name} is in #{user.location}."` and
+then make the location options "the yard", "the kitchen", "the bathroom" or "the
+bedroom", you _will_ sort out that particular sentence (as far as I know, but
+see below). But consider something like `"Your friend is wearing a #{garmet.color} #{garmet.type}."`
+Many languages expect adjectives to agree on gender with the noun they modify.
+If your friend could be wearing a shirt or a hat that can be red or purple, you
+need two of each color and need to be aware of which garmet needs which version
+of the appropriate color.
+
+
+## I can write enough grammatical metadata to be able to stitch togeter coherent sentences in all the languages I'm targeting.
+
+There are whole research projects by leading linguists and computer scientists
+working on this problem. Solving this in general is _ridiculously_ hard, it
+seems. Even solving it for one specific pair of languages requires you to know a
+ton about the grammatical rules of both languages and most native English
+speakers cannot express the grammar rules they routinely adhere to on the fly as
```

Comment:
This made me think of [adjective order](https://dictionary.cambridge.org/us/grammar/british-grammar/about-adjectives-and-adverbs/adjectives-order), which would be a fun link